packages/core/core/src/RequestTracker.js

```diff
@@ -1108,6 +1117,74 @@ export default class RequestTracker {
     return {api, subRequestContentKeys};
   }
 
+  generateInvalidationReport(): void {
+    let nodes = {};
+    let invalidationRelations = {};
+    for (let {invalidated, cause} of this.graph.invalidationReport) {
+      let invalidatedNode = this.graph.getNode(invalidated);
+      let causeNode =
+        cause !== undefined ? this.graph.getNode(cause) : undefined;
+      if (
+        invalidatedNode === null ||
+        causeNode === null ||
+        invalidatedNode === undefined ||
+        causeNode === undefined
+      ) {
+        continue;
+      }
+
+      if (invalidatedNode.id !== 'Main') {
+        if (!invalidationRelations[invalidated]) {
+          invalidationRelations[invalidated] = [];
+        }
+
+        if (!invalidationRelations[invalidated].includes(cause)) {
+          invariant(cause !== undefined);
+          invalidationRelations[invalidated].push(cause);
+        }
+      }
+
+      invariant(invalidatedNode.type === 1);
+      nodes[invalidated] = {
+        id: invalidatedNode.id,
+        type: invalidatedNode.type,
+        requestType: invalidatedNode.requestType,
+        invalidateReason: invalidatedNode.invalidateReason,
+      };
+
+      invariant(cause !== undefined);
+      if (
+        causeNode.requestType !== undefined &&
+        causeNode.invalidateReason !== undefined
+      ) {
+        nodes[cause] = {
+          id: causeNode.id,
+          type: causeNode.type,
+          requestType: causeNode.requestType,
+          invalidateReason: causeNode.invalidateReason,
+        };
+      } else {
+        nodes[cause] = {
+          id: causeNode.id,
+          type: causeNode.type,
+        };
+      }
+    }
+
+    const fs = require('fs');
```

Comment:
Parcel has the ability to override the output filesystem which this ignores. Would it be hard to use `outputFS` instead of requiring the node FS directly?